perses/samplers/samplers.py

```diff
@@ -36,12 +41,6 @@
 import logging
 logger = logging.getLogger(__name__)
 
-################################################################################
-# THERMODYNAMIC STATE
-################################################################################
-
-from perses.samplers.thermodynamics import ThermodynamicState
-
 ################################################################################
 # UTILITY FUNCTIONS
 ################################################################################
```

Comment:
You could also replace `log_sum_exp` with [`scipy.misc.logsumexp`](https://docs.scipy.org/doc/scipy/reference/generated/scipy.misc.logsumexp.html).

Also, it seems like the implementation of `log_sum_exp` below does not compute it properly! It neglects to add back `a_n.max()` before returning the result!

Reply:
Oh, whoops! I will just use the scipy one then.